apps/dashboard/test/models/batch_connect/session_context_test.rb

```diff
@@ -0,0 +1,109 @@
+  #   end
+  # end
+  #
+  # test "should not update cache if per app cache disabled" do
+  #   app = BatchConnect::App.new(router: nil)
+  #   app.stubs(:form_config).returns(cacheable: false, attributes: { num_cores: { widget: "number_field", value: "1" } }, form: ["bc_account", "num_cores"])
+  #   context = app.build_session_context
+  #
+  #   context.update_with({"bc_account" => "PZS0714", "num_cores" => "28"})
+  #
+  #   assert_equal ["", "1"], [context["bc_account"].value, context["num_cores"].value]
+  # end
+  #
+  # test "should not update single field if per attribute cache disabled" do
+  #   app = BatchConnect::App.new(router: nil)
+  #   app.stubs(:form_config).returns(attributes: { num_cores: { widget: "number_field", value: "1", cacheable: false } }, form: ["bc_account", "num_cores"])
+  #   context = app.build_session_context
+  #
+  #   context.update_with({"bc_account" => "PZS0714", "num_cores" => "28"})
+  #
+  #   assert_equal ["PZS0714", "1"], [context["bc_account"].value, context["num_cores"].value]
+  # end
+  #
+  # test "should not update single field if per attribute cache disabled even if global cache setting disabled but per app cache enabled" do
+  #   with_modified_env(OOD_BATCH_CONNECT_CACHE_ATTR_VALUES: 'FALSE') do
+  #     app = BatchConnect::App.new(router: nil)
+  #     app.stubs(:form_config).returns(cacheable: true, attributes: { num_cores: { widget: "number_field", value: "1", cacheable: false } }, form: ["bc_account", "num_cores"])
+  #     context = app.build_session_context
+  #
+  #     context.update_with({"bc_account" => "PZS0714", "num_cores" => "28"})
+  #
+  #     assert_equal ["PZS0714", "1"], [context["bc_account"].value, context["num_cores"].value]
+  #   end
+  # end
+  #
+  # test "should update single field if per attribute cache enabled even if global cache setting disabled" do
+  #   with_modified_env(OOD_BATCH_CONNECT_CACHE_ATTR_VALUES: 'FALSE') do
+  #     app = BatchConnect::App.new(router: nil)
+  #     app.stubs(:form_config).returns(attributes: { num_cores: { widget: "number_field", value: "1", cacheable: true } }, form: ["bc_account", "num_cores"])
+  #     context = app.build_session_context
+  #
+  #     context.update_with({"bc_account" => "PZS0714", "num_cores" => "28"})
+  #
+  #     assert_equal ["", "28"], [context["bc_account"].value, context["num_cores"].value]
+  #   end
+  # end
+  #
+  # test "should update single field if per attribute cache enabled even if per app cache disabled" do
+  #   app = BatchConnect::App.new(router: nil)
+  #   app.stubs(:form_config).returns(cacheable: false, attributes: { num_cores: { widget: "number_field", value: "1", cacheable: true } }, form: ["bc_account", "num_cores"])
+  #   context = app.build_session_context
+  #
+  #   context.update_with({"bc_account" => "PZS0714", "num_cores" => "28"})
+  #
+  #   assert_equal ["", "28"], [context["bc_account"].value, context["num_cores"].value]
+  # end
```

Comment:
If these tests all pass now, go ahead and un-comment these. If a method name needs updated (i.e. `update_with` => `update_with_cache`) please do so.